src/Plugin/tmgmt/Translator/ThebigwordTranslator.php

```diff
@@ -0,0 +1,872 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\tmgmt_thebigword\Plugin\tmgmt\Translator\ThebigwordTranslator.
+ */
+
+namespace Drupal\tmgmt_thebigword\Plugin\tmgmt\Translator;
+
+use Drupal\Core\Cache\Cache;
+use Drupal\Core\Datetime\DrupalDateTime;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\StringTranslation\TranslatableMarkup;
+use Drupal\Core\Url;
+use Drupal\tmgmt\ContinuousTranslatorInterface;
+use Drupal\tmgmt\Entity\JobItem;
+use Drupal\tmgmt\Entity\RemoteMapping;
+use Drupal\tmgmt\Entity\Translator;
+use Drupal\tmgmt\JobItemInterface;
+use Drupal\tmgmt\RemoteTranslatorInterface;
+use Drupal\tmgmt\SourcePreviewInterface;
+use Drupal\tmgmt\TMGMTException;
+use Drupal\tmgmt\TranslatorPluginBase;
+use GuzzleHttp\Exception\BadResponseException;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use GuzzleHttp\ClientInterface;
+use Drupal\tmgmt\TranslatorInterface;
+use Drupal\tmgmt\JobInterface;
+use Drupal\tmgmt\Translator\AvailableResult;
+
+/**
+ * Thebigword translation plugin controller.
+ *
+ * @TranslatorPlugin(
+ *   id = "thebigword",
+ *   label = @Translation("Thebigword translator"),
+ *   description = @Translation("Thebigword translator service."),
+ *   ui = "Drupal\tmgmt_thebigword\ThebigwordTranslatorUi"
+ * )
+ */
+class ThebigwordTranslator extends TranslatorPluginBase implements ContainerFactoryPluginInterface, ContinuousTranslatorInterface, RemoteTranslatorInterface {
+
+  /**
+   * Translation service URL.
+   */
+  const PRODUCTION_URL = 'http://uat-integration.thebigword.com/drupal/api';
```

Comment:
maybe this would be better to have in config, without UI. instead of the mock flag, that we don't use right now afaik anyway.